draft-ietf-quic-transport.md

```diff
@@ -4697,11 +4697,14 @@ active_connection_id_limit (0x000e):
   to store. This value includes the connection ID received during the handshake,
   that received in the preferred_address transport parameter, and those received
   in NEW_CONNECTION_ID frames.
-  Unless a zero-length connection ID is being used, the value of the
-  active_connection_id_limit parameter MUST be no less than 2. If this
-  transport parameter is absent, a default of 2 is assumed.
-  When a zero-length connection ID is being used, the active_connection_id_limit
-  parameter MUST NOT be sent.
+  The value of the active_connection_id_limit parameter MUST be at least 2.
+  An endpoint that receives a value less than 2 MUST close the connection
+  with an error of type TRANSPORT_PARAMETER_ERROR.
+  If this transport parameter is absent, a default of 2 is assumed.  If an
+  endpoint uses a zero-length connection ID, the active_connection_id_limit
```

Comment:
"uses" here seems confusing

If an endpoint issues a zero-length connection-id, the active_connection_id_limit value it says is ignored, as...

Reply:
Here's the situation I'm trying to explain:

Suppose the client (C) uses 0-length CIDs. Since C uses 0-length CIDs, it will never send a NEW_CONNECTION_ID frame. The server (S) can still send an active_connection_id_limit transport param, but C will ignore it because it is irrelevant.

> If an endpoint issues a zero-length connection-id, the active_connection_id_limit value it says is ignored, as...

My interpretation of this is that it says if a client (for example) C uses the 0-length CID, the active_connection_id_limit sent by C should be ignored. I believe this is incorrect, as C's active_connection_id_limit is regarding how many NEW_CONNECTION_IDs S can send to C.

Reply:
My problem here is the text "uses", which is non-specific. A issues a CID and B sends packets with it. So they both use it.

You're right that my text above is backwards, which seems like an argument in favor of "this is confusing text".  How about:

"If an endpoint issues a zero-length connection ID, it will never send NEW_CONNECTION_ID and therefore ignores any active_connection_id limits values it receives"

Reply:
I've changed "uses" to "issues" and tightened up the language.